### ISSUE TITLE
Problem when trying to use theme from vscode marketplace

### ISSUE THREAD

**Issue by leomotta121** [andrewbranch/gatsby-remark-vscode#34]
This error is when I try to fetch the html code from markdown.

Here is my gatsby-config.js:

```javascript
'use strict';

const siteConfig = require('./config.js');
const postCssPlugins = require('./postcss-config.js');

module.exports = {
  pathPrefix: siteConfig.pathPrefix,
  siteMetadata: {
    url: siteConfig.url,
    title: siteConfig.title,
    subtitle: siteConfig.subtitle,
    copyright: siteConfig.copyright,
    disqusShortname: siteConfig.disqusShortname,
    menu: siteConfig.menu,
    author: siteConfig.author
  },
  plugins: [
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: `${__dirname}/content`,
        name: 'pages'
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: `${__dirname}/static/media`,
        name: 'media'
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'css',
        path: `${__dirname}/static/css`
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        name: 'assets',
        path: `${__dirname}/static`
      }
    },
    {
      resolve: 'gatsby-plugin-feed',
      options: {
        query: `
          {
            site {
              siteMetadata {
                site_url: url
                title
                description: subtitle
              }
            }
          }
        `,
        feeds: [{
          serialize: ({ query: { site, allMarkdownRemark } }) => (
            allMarkdownRemark.edges.map((edge) => Object.assign({}, edge.node.frontmatter, {
              description: edge.node.frontmatter.description,
              date: edge.node.frontmatter.date,
              url: site.siteMetadata.site_url + edge.node.fields.slug,
              guid: site.siteMetadata.site_url + edge.node.fields.slug,
              custom_elements: [{ 'content:encoded': edge.node.html }]
            }))
          ),
          query: `
              {
                allMarkdownRemark(
                  limit: 1000,
                  sort: { order: DESC, fields: [frontmatter___date] },
                  filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
                ) {
                  edges {
                    node {
                      html
                      fields {
                        slug
                      }
                      frontmatter {
                        title
                        date
                        template
                        draft
                        description
                      }
                    }
                  }
                }
              }
            `,
          output: '/rss.xml'
        }]
      }
    },
    {
      resolve: 'gatsby-transformer-remark',
      options: {
        plugins: [
          'gatsby-remark-relative-images',
          {
            resolve: 'gatsby-remark-katex',
            options: {
              strict: 'ignore'
            }
          },
          {
            resolve: 'gatsby-remark-images',
            options: {
              maxWidth: 960,
              withWebp: true,
              ignoreFileExtensions: [],
            }
          },
          {
            resolve: 'gatsby-remark-responsive-iframe',
            options: { wrapperStyle: 'margin-bottom: 1.0725rem' }
          },
          {
            resolve: 'gatsby-remark-vscode',
            options: {
              extensions: [{
                identifier: 'RobbOwen.synthwave-vscode',
                version: '0.0.7'
              }]
            }
          },
          'gatsby-remark-autolink-headers',
          'gatsby-remark-copy-linked-files',
          'gatsby-remark-smartypants',
          'gatsby-remark-external-links'
        ]
      }
    },
    'gatsby-transformer-sharp',
    'gatsby-plugin-sharp',
    'gatsby-plugin-netlify',
    {
      resolve: 'gatsby-plugin-netlify-cms',
      options: {
        modulePath: `${__dirname}/src/cms/index.js`,
      }
    },
    {
      resolve: 'gatsby-plugin-google-gtag',
      options: {
        trackingIds: [siteConfig.googleAnalyticsId],
        pluginConfig: {
          head: true,
        },
      },
    },
    {
      resolve: 'gatsby-plugin-sitemap',
      options: {
        query: `
          {
            site {
              siteMetadata {
                siteUrl: url
              }
            }
            allSitePage(
              filter: {
                path: { regex: "/^(?!/404/|/404.html|/dev-404-page/)/" }
              }
            ) {
              edges {
                node {
                  path
                }
              }
            }
          }
        `,
        output: '/sitemap.xml',
        serialize: ({ site, allSitePage }) => allSitePage.edges.map((edge) => ({
          url: site.siteMetadata.siteUrl + edge.node.path,
          changefreq: 'daily',
          priority: 0.7
        }))
      }
    },
    {
      resolve: 'gatsby-plugin-manifest',
      options: {
        name: siteConfig.title,
        short_name: siteConfig.title,
        start_url: '/',
        background_color: '#FFF',
        theme_color: '#F7A046',
        display: 'standalone',
        icon: 'static/photo.jpg'
      },
    },
    'gatsby-plugin-offline',
    'gatsby-plugin-catch-links',
    'gatsby-plugin-react-helmet',
    {
      resolve: 'gatsby-plugin-sass',
      options: {
        postCssPlugins: [...postCssPlugins],
        cssLoaderOptions: {
          camelCase: false,
        }
      }
    },
    'gatsby-plugin-flow',
    'gatsby-plugin-optimize-svgs',
  ]
};
```

Here is the error I am getting:

```
The GraphQL query from /home/leo/Desktop/projects/blog/src/templates/post-template.js failed.

Errors:
  Unexpected token }

  GraphQL request:6:5
  5 |     id
  6 |     html
    |     ^
  7 |     fields {
URL path:
  /posts/a-brief-history-of-typography/
Context:
  {
    "slug": "/posts/a-brief-history-of-typography/"
  }
Plugin:
  none
Query:
  query PostBySlug(
    $slug: String!
  ) {
    markdownRemark(fields: {slug: {eq: $slug}}) {
      id
      html
      fields {
        slug
        tagSlugs
      }
      frontmatter {
        date
        description
        tags
        title
      }
    }
  }
```
If I try to use a supported theme such as Dark+ it works fine.

**Comment by andrewbranch** [comment]
Hey @leomotta121! As it turns out, that theme contains invalid JSON:

<img width="878" alt="image" src="https://user-images.githubusercontent.com/3277153/62796051-06474580-ba8d-11e9-9cba-1ef841d2588e.png">

I’m pretty sure I’m parsing themes as “JSON with comments,” but it must not support trailing commas. I’ll look into enabling that since apparently VS Code itself is fine with it.

**Comment by andrewbranch** [comment]
Published in v1.1.1

**Comment by brotzky** [comment]
I've run into a similar issue, but it might be more related to `gatsby-plugin-mdx` than this project.

```js
    {
      resolve: `gatsby-plugin-mdx`,
      options: {
        extensions: [`.mdx`, `.md`],
        gatsbyRemarkPlugins: [
          {
            resolve: `gatsby-remark-vscode`,
            options: {
              extensions: [
                {
                  identifier: `naumovs.theme-oceanicnext`,
                  version: `1.1.5`,
                },
              ],
            },
          },
        ]
     }
```

>  ERROR #11321  PLUGIN
> "gatsby-plugin-mdx" threw an error while running the onCreateNode lifecycle:
> JSON5: invalid character '<' at 1:1
>  SyntaxError: JSON5: invalid character '<' at 1:1

Without defining a custom extension the plugin works great.

**Comment by andrewbranch** [comment]
Oh, it’s because that theme is published in XML instead of JSON. I could have sworn I already handled that, but apparently not in this code path.

**Comment by brotzky** [comment]
Ah, cool! Glad I chose the correct report this issue in. Thanks for investigating @andrewbranch!

**Comment by leomotta121** [comment]
Sorry for my late answer, yes, I did manage to do it with other themes, would like to use it, because I was planning my blog based on this theme, but I am switching to Dracula and it works fine :D

**Comment by andrewbranch** [comment]
@leomotta121 your case with synthwave is now fixed in v1.1.1, but @brotzky’s oceanic-next issue is a different but related problem that I recycled this same issue for (hopefully that didn’t cause too much confusion). It shouldn’t be much trouble to fix XML themes, I’ve just been swamped the past week 😅

**Comment by leomotta121** [comment]
I have not yet tested, I will try it when I arrive at home, but sounds exciting! SynthWave is really beautiful and the only implementation I found form prismjs does not look really good, thanks for the attention!

**Comment by brotzky** [comment]
Thanks @andrewbranch. No rush from me. Do it when you're relaxed and have the time. Enjoy yourself :)

**Comment by andrewbranch** [comment]
Fixed in v1.2.0

**Comment by leomotta121** [comment]
@andrewbranch works fine here!
